qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -444,10 +444,308 @@ def particle_number(orbitals, mapping="jordan_wigner", wires=None):
+        for i in range(num_indices)
+        if all(indices[j][i] in active for j in range(len(indices)))
+    ]
+
+    # Building the table of indices and matrix elements
+    t_table = np.zeros((2 * len(pairs), 3))
+
+    for i, pair in enumerate(pairs):
+        alpha, beta = pair
+        element = t_matrix_elements[alpha, beta]
+
+        # spin-up term
+        t_table[2 * i, 0] = 2 * active.index(alpha)
+        t_table[2 * i, 1] = 2 * active.index(beta)
+        t_table[2 * i, 2] = element
+
+        # spin-down term
+        t_table[2 * i + 1, 0] = 2 * active.index(alpha) + 1
+        t_table[2 * i + 1, 1] = 2 * active.index(beta) + 1
+        t_table[2 * i + 1, 2] = element
+
+    return t_table, t_core
+
+
+def two_particle(v_matrix_elements, core=None, active=None, cutoff=1.0e-12):
+    r"""Generates the table of matrix elements of a given two-particle operator
+    required to build many-body qubit observables.
+
+    Second quantized two-particle operators are expanded in the basis of single-particle
+    states as
+
+    .. math::
+
+        \hat{V} = \frac{1}{2} \sum_{\alpha, \beta, \gamma, \delta}
+        \langle \alpha, \beta \vert \hat{v} \vert \gamma, \delta \rangle
+        ~ &[& \hat{c}_{\alpha\uparrow}^\dagger \hat{c}_{\beta\uparrow}^\dagger
+        \hat{c}_{\gamma\uparrow} \hat{c}_{\delta\uparrow} + \hat{c}_{\alpha\uparrow}^\dagger
+        \hat{c}_{\beta\downarrow}^\dagger \hat{c}_{\gamma\downarrow} \hat{c}_{\delta\uparrow} \\
+        &+& \hat{c}_{\alpha\downarrow}^\dagger \hat{c}_{\beta\uparrow}^\dagger
+        \hat{c}_{\gamma\uparrow} \hat{c}_{\delta\downarrow} + \hat{c}_{\alpha\downarrow}^\dagger
+        \hat{c}_{\beta\downarrow}^\dagger \hat{c}_{\gamma\downarrow} \hat{c}_{\delta\downarrow}~].
+
+    In the equation above the indices :math:`\alpha, \beta, \gamma, \delta` run over the basis
+    of spatial orbitals :math:`\vert \alpha \rangle = \phi_\alpha(r)`. Since the operator
+    :math:`v` acts only on the spatial the spin quantum numbers are indicated explicitly
+    with the up/down arrows. The operators :math:`\hat{c}^\dagger` and :math:`\hat{c}`
+    are the particle creation and annihilation operators, respectively, and
+    :math:`\langle \alpha, \beta \vert \hat{v} \vert \gamma, \delta \rangle` denotes the
+    matrix elements of the operator :math:`\hat{v}`
+
+    .. math::
+
+        \langle \alpha, \beta \vert \hat{v} \vert \gamma, \delta \rangle =
+        \int dr_1 \int dr_2 ~ \phi_\alpha^*(r_1) \phi_\beta^*(r_2) ~\hat{v}(r_1, r_2)~
+        \phi_\gamma(r_2) \phi_\delta(r_1).
+
+    If an active space is defined (see :func:`~.active_space`), the summation indices
+    run over the active orbitals and the contribution due to core orbitals, is computed as
+    :math:`V_\mathrm{core} = 2 \sum_{\alpha, \beta \in \mathrm{core}}
+    \langle \alpha, \beta \vert \hat{v} \vert \alpha, \beta \rangle`.
+
+    Args:
+        v_matrix_elements (array[float]): 4D NumPy array with the matrix elements
```

Comment:
One question for me is: where is the user supposed to get access to these matrix elements? Is this a standard output from quantum chemistry packages?

Reply:
Yes. It is not straightforward to get them printed out but there is always a possibility. In the particular cases of PySCF and Psi4 this is straightforward thanks to the interfaces with OpenFermion.